### PR TITLE
Use `retry` convention over calling the method manually.

### PR DESCRIPTION
**Pull request by minuscorp**

### CODE REVIEW

File: features/project/project_controller.rb [hunk 1]
```diff
@@ -70,11 +70,32 @@ class ProjectController < AuthenticatedControllerBase
...
+
+      fastfile.all_lanes_flat.to_json
+    end
+
     get "#{HOME}/add/*" do |repo_name|
       provider_credential = check_and_get_provider_credential(type: FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github])
 
       github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
-      selected_repo = github_service.repos.select { |repo| repo_name == repo.name }.first
+      selected_repo = github_service.repos.detect { |repo| repo_name == repo.name }
```
Comment (taquitos) [inline features/project/project_controller.rb]: til: `.detect`

File: shared/models/git_repo.rb [hunk 2]
```diff
@@ -118,20 +118,17 @@ def initialize(git_config: nil, local_folder: nil, provider_credential: nil, asy
...
           @_git = Git.open(self.local_folder)
         rescue ArgumentError => aex
-          if tries >= 1
-            raise aex
-          end
-
           logger.debug("Path #{self.local_folder} is not a git directory, deleting and trying again")
           self.clear_directory
-          return self.setup_repo(tries: tries + 1)
+          retry if (retry_count += 1) < 5
```
Comment (taquitos) [inline shared/models/git_repo.rb]: til: `retry`

File: features/project/project_controller.rb [hunk 3]
```diff
@@ -70,11 +70,32 @@ class ProjectController < AuthenticatedControllerBase
...
+
+      provider_credential = check_and_get_provider_credential(type: FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github])
+
+      selected_repo = github_service.repos.detect { |repo| repo_name == repo.name }
+      repo_config = GitRepoConfig.from_octokit_repo!(repo: org + "/" + repo_name)
+
+      git_repo = GitRepo.new(
+        git_config: repo_config,
+        provider_credential: provider_credential,
+        local_folder: dir,
```
Comment (taquitos) [inline features/project/project_controller.rb]: where is `dir` coming from?

**Comment by minuscorp** [comment]
My git tree messed up. Opened new...

**Pull request closed without merging**